htdocs/core/class/smtps.class.php

```diff
@@ -373,14 +388,31 @@ private function _server_connect()
             $this->_setErr(99, $host.' is either offline or is an invalid host name.');
             $_retVal = false;
         } else {
-            //See if we can connect to the SMTP server
-            if ($this->socket = @fsockopen(
-                preg_replace('@tls://@i', '', $this->getHost()), // Host to 'hit', IP or domain
+            if (function_exists('stream_socket_client')) {
+                $socket_context = stream_context_create($this->_options); // An array of options for stream_context_create()
+                set_error_handler([$this, 'errorHandler']);
+                $this->socket = @stream_socket_client(
+                    $this->getHost() .    // Host to 'hit', IP or domain
+                    ':' . $this->getPort(),             // which Port number to use
+                    $this->errno,                       // actual system level error
+                    $this->errstr,                      // and any text that goes with the error
+                    $this->_smtpTimeout,                // timeout for reading/writing data over the socket
+                    STREAM_CLIENT_CONNECT,
+                    $socket_context                     // Options for connection
+                );
+        	} else {
+                $this->socket = @fsockopen(
+                    $this->getHost(),       // Host to 'hit', IP or domain
```

Comment:
Old code for this line was 
preg_replace('@tls://@i', '', $this->getHost()), // Host to 'hit', IP or domain
Is it done on purpose that your remove the clean of the Host ? (If my memory is good, having the tls:// in this parameter generates errors with outlook or office 365 that except to have a hostname only)

Reply:
It's a error. It's corrected